app/controllers/carts_controller.rb

```diff
@@ -7,4 +8,19 @@ def show
     @show_comments = true
   end
 
+  def index
+    @role = 'requester'
+    my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
+    if my_carts.empty?
+      @role = 'approver'
+      my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
+    end
```

Comment:
This seems over-complicated code and UX-wise... can we just do this?

```ruby
my_carts = current_user.carts.order('created_at DESC')
```

We can [do a fancier query](http://stackoverflow.com/questions/6332043/sql-order-by-multiple-values-in-specific-order) if we want to show carts where the user is a requester before those where they're an approver, but

* I don't think there will be a lot of overlap between requesters and approvers, at least for WHSC
* My instinct is that the user would want to see all carts they're involved with in one list anyway